pyseir/cli.py

```diff
@@ -52,114 +56,190 @@ def _generate_whitelist() -> pd.DataFrame:
+        # and parameters used to select subsets of regions.
+        whitelist_regions = set(
+            whitelist.regions_in_states(
+                list(state_fitter_map.keys()), fips=fips, whitelist_df=whitelist_df
+            )
+        )
+        pipeline_inputs = [
+            SubStateRegionPipelineInput(
+                region=region,
+                run_fitter=(region in whitelist_regions),
+                state_fitter=state_fitter_map.get(region.get_state_region()),
+                regional_combined_dataset=pipeline.RegionalCombinedData.from_region(region),
+            )
+            for region in (infer_rt_regions | whitelist_regions)
+        ]
+        return pipeline_inputs
+
+
+@dataclass
+class SubStatePipeline:
+    """Runs the pipeline for one region smaller than a state and stores the output."""
+
+    region: pipeline.Region
+    infer_df: pd.DataFrame
+    _combined_data: pipeline.RegionalCombinedData
+    fitter: Optional[model_fitter.ModelFitter] = None
+    ensemble: Optional[ensemble_runner.EnsembleRunner] = None
+
+    @staticmethod
+    def run(input: SubStateRegionPipelineInput) -> "SubStatePipeline":
+        assert not input.region.is_state()
+        # `infer_df` does not have the NEW_ORLEANS patch applied. TODO(tom): Rename to something like
+        # infection_rate.
+        infer_rt_input = infer_rt.RegionalInput.from_region(input.region)
+        infer_df = infer_rt.run_rt(infer_rt_input)
+        if input.run_fitter:
+            fitter_input = model_fitter.RegionalInput.from_substate_region(
+                input.region, input.state_fitter
+            )
+            fitter = model_fitter.ModelFitter.run_for_region(fitter_input)
+            ensembles_input = ensemble_runner.RegionalInput.for_substate(
+                fitter, state_fitter=input.state_fitter
+            )
+            ensemble = ensemble_runner.make_and_run(ensembles_input)
+        else:
+            fitter = None
+            ensemble = None
+        return SubStatePipeline(
+            region=input.region,
+            infer_df=infer_df,
+            fitter=fitter,
+            ensemble=ensemble,
+            _combined_data=input.regional_combined_dataset,
+        )
+
+    def population(self) -> float:
+        return self._combined_data.get_us_latest()[CommonFields.POPULATION]
 
-    with Pool(maxtasksperchild=10) as p:
-        results = pd.concat(p.map(infer_rt.run_rt, regions))
 
-    if results.empty:
-        return pd.DataFrame()
+def _patch_substatepipeline_nola_infection_rate(
+    input_pipelines: List[SubStatePipeline],
+) -> List[SubStatePipeline]:
+    """Returns a new list of pipeline objects with New Orleans infection rate patched."""
+    pipeline_map = {p.region.fips: p for p in input_pipelines}
 
-    need_patch = set(results.fips) & set(NEW_ORLEANS_FIPS)
+    input_fips = set(pipeline_map.keys())
+    need_patch = input_fips & set(NEW_ORLEANS_FIPS)
```

Comment:
super small, `fips_to_patch` is more clear to me reading through

Reply:
Done